packages/web/src/components/createCell.tsx

```diff
@@ -77,6 +77,12 @@ type Guaranteed<T> = {
   [K in keyof T]-?: NonNullable<T[K]>
 }
 
+type KeyCount<T extends object> = L.Length<U.ListOf<O.SelectKeys<T, any>>>
+
+type ConditionallyGuaranteed<T extends object> = KeyCount<T> extends 1
+  ? Guaranteed<T>
+  : T
```

Comment:
Here's how we decide whether to guarantee the data.  With exactly one key, we guarantee, else we don't.